pkg/api/core/v1/client/http.go

```diff
@@ -36,6 +36,18 @@ func wrapResponseError(err error, code int) *responseError {
 	return &responseError{error: err, statusCode: code}
 }
 
+func (c *Client) DisableVersionWarning() {
+	value := false
+	c.versionWarning = &value
+}
+
+// VersionWarningEnabled returns true if versionWarning field is either not
+// set of true. That makes "true" the default value unless DisableVersionWarning
+// is called.
+func (c *Client) VersionWarningEnabled() bool {
+	return c.versionWarning == nil || *c.versionWarning
```

Comment:
Name the field `noVersionWarning`, with a default of `false`. `Disable...` then simply sets it to `true`.
That gets rid of all the gymnastics with a pointer. Just a plain bool.

Reply:
I'm not sure which one I dislike more, double negations or the pointer hacks. I'll try your suggestion.